### PR TITLE
ref: Add types to store and symbolication

### PR DESCRIPTION
**Pull request by loewenheim**
This adds type checking to `tasks/store.py` and `tasks/symbolication.py`. The only place where I'm not sure I did the right thing is `create_failed_event`; I gave `issues` and `reprocessing_rev` the types `List[Dict[str, str]]` and `Any`, respectively.

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -381,13 +405,18 @@ def _continue_to_save_event():
...
     queue="events.process_event",
     time_limit=65,
     soft_time_limit=60,
 )
-def process_event(cache_key, start_time=None, event_id=None, data_has_changed=None, **kwargs):
+def process_event(
+    cache_key: Optional[str],
+    start_time: Optional[int] = None,
+    event_id: Optional[str] = None,
+    data_has_changed: bool = False,
```
Comment (untitaker) [inline src/sentry/tasks/store.py:418]: I don't think removing `**kwargs` is safe here. Do we have any invocations of `process_event` that pass more arguments? If yes they need to be removed in a separate PR and deployed separately.

File: src/sentry/tasks/store.py [hunk 2]
```diff
@@ -696,13 +740,17 @@ def time_synthetic_monitoring_event(data, project_id, start_time):
...
 )
 def save_event(
-    cache_key=None, data=None, start_time=None, event_id=None, project_id=None, **kwargs
-):
-    _do_save_event(cache_key, data, start_time, event_id, project_id, **kwargs)
+    cache_key: Optional[str] = None,
+    data: Optional[Any] = None,
+    start_time: Optional[int] = None,
+    event_id: Optional[str] = None,
+    project_id: Optional[int] = None,
```
Comment (untitaker) [inline src/sentry/tasks/store.py:754]: same comment about `**kwargs` here

File: src/sentry/tasks/store.py [hunk 3]
```diff
@@ -407,15 +436,18 @@ def process_event(cache_key, start_time=None, event_id=None, data_has_changed=No
...
     time_limit=65,
     soft_time_limit=60,
 )
 def process_event_from_reprocessing(
-    cache_key, start_time=None, event_id=None, data_has_changed=None, **kwargs
-):
+    cache_key: Optional[str],
+    start_time: Optional[int] = None,
+    event_id: Optional[str] = None,
+    data_has_changed: bool = False,
```
Comment (untitaker) [inline src/sentry/tasks/store.py:449]: note to self, snooped around a bit in the code and this type change seems safe

File: src/sentry/tasks/symbolication.py [hunk 4]
```diff
@@ -220,37 +242,45 @@ def _continue_to_process_event():
...
-@instrumented_task(
+@instrumented_task(  # type: ignore
     name="sentry.tasks.store.symbolicate_event",
     queue="events.symbolicate_event",
     time_limit=settings.SYMBOLICATOR_PROCESS_EVENT_HARD_TIMEOUT + 30,
     soft_time_limit=settings.SYMBOLICATOR_PROCESS_EVENT_HARD_TIMEOUT + 20,
     acks_late=True,
 )
-def symbolicate_event(cache_key, start_time=None, event_id=None, **kwargs):
+def symbolicate_event(
```
Comment (untitaker) [inline src/sentry/tasks/symbolication.py:252]: Same concern here

File: src/sentry/tasks/symbolication.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+    project_id = data["project"]
+    set_current_event_project(project_id)
+
+    event_id = data["event_id"]
+
+    from_reprocessing = symbolicate_task is symbolicate_event_from_reprocessing
+
+    def _continue_to_process_event() -> None:
+        process_task = process_event_from_reprocessing if from_reprocessing else process_event
+        # TODO: this uses a private "store" function. is there a way to not do this?
```
Comment (relaxolotl) [inline src/sentry/tasks/symbolication.py:108]: nts: figure out an answer to this, namely "is this ok?" and how to make it ok if it isn't

also probably good to think about the the line that has been drawn by splitting all of the symbolicator stuff out into its own file: are there other places where it's using internal `store.py` stuff? is there symbolication logic in `store.py` that could and should be moved out into a method in this file? what are the low hanging fruit for those?

Comment (relaxolotl) [reply src/sentry/tasks/symbolication.py:108]: I LEFT THIS ON THE WRONG PR SOMEHOW I'M SORRY

**Review by Swatinem — COMMENTED** [review]
lgtm from my limited understanding of this. I do have two Q however:
* is removal of `submit_process/project` really safe? I saw one callsite that was updated. is that all?
* what @untitaker sayd about `kwargs`. I know way too little about how they really work.

**Comment by loewenheim** [comment]
Thank you for your advice about those `Any`s, @evanh! The only remaining ones are in `**kwargs: Dict[str, Any]`, which I don't think can be improved upon, and `reprocessing_rev: Any`, which probably has a more precise type, but I can't for the life of me find out what it is.

File: src/sentry/tasks/store.py [hunk 6]
```diff
@@ -376,13 +409,19 @@ def _continue_to_save_event():
...
     time_limit=65,
     soft_time_limit=60,
 )
-def process_event(cache_key, start_time=None, event_id=None, data_has_changed=None, **kwargs):
+def process_event(
+    cache_key: Optional[str],
+    start_time: Optional[int] = None,
+    event_id: Optional[str] = None,
+    data_has_changed: bool = False,
+    **kwargs: Dict[str, Any],
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply src/sentry/tasks/store.py:423]: Oh, I didn't realize that. I'll fix it.

File: src/sentry/tasks/symbolication.py [hunk 7]
```diff
@@ -193,7 +210,12 @@ def _continue_to_process_event():
                             "tasks.store.symbolicate_event.retry",
                             tags={"symbolication_function": symbolication_function_name},
                         )
-                        sleep(min(e.retry_after, SYMBOLICATOR_MAX_RETRY_AFTER))
+                        sleep_time = (
+                            min(e.retry_after, SYMBOLICATOR_MAX_RETRY_AFTER)
+                            if e.retry_after
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply src/sentry/tasks/symbolication.py:215]: Sigh. I got accustomed to checking if a value is `None` by doing `if value`.

**Review by evanh — APPROVED** [review]
LGTM, one question though.

**Review by flub — APPROVED** [review]
